docs/manual/resource-leak-checker.tex

```diff
@@ -588,6 +597,234 @@
 will only use \<@MustCallUnknown> as a default when encountering a language
 feature that it is unable to reason about.
 
+\sectionAndLabel{Arrays of resources}{resource-leak-arrays}
+To understand how the checker handles arrays, let us first remind ourselves what happens if we assign resources with non-empty must-call type to an array:
+
+\begin{verbatim}
+Socket[] s = new Socket[n];
+s[0] = new Socket(myHost, myPort);
+\end{verbatim}
+
+The assignment throws an error, because the scope of the constructed Socket ends, without its must-call obligations having been fulfilled.
+
+\begin{verbatim}
+Example.java:31: error: [required.method.not.called] @MustCall method close may not have been invoked on new Socket(myHost, myPort) or any of its aliases.
+            arr[0] = new Socket(myHost, myPort);
```

Comment:
Should it be `s[0]`?  Or should the `Socket[]` in the example above be named `arr`?

Reply:
you're right, it should be s[0], I corrected it.